src/Reflection/ReflectionProperty.php

```diff
@@ -592,9 +671,123 @@ private function computeModifiers(PropertyNode $node): int
         $modifiers  = $node->isReadonly() ? ReflectionPropertyAdapter::IS_READONLY : 0;
         $modifiers += $node->isStatic() ? CoreReflectionProperty::IS_STATIC : 0;
         $modifiers += $node->isPrivate() ? CoreReflectionProperty::IS_PRIVATE : 0;
+        $modifiers += $node->isPrivateSet() ? ReflectionPropertyAdapter::IS_PRIVATE_SET_COMPATIBILITY : 0;
         $modifiers += $node->isProtected() ? CoreReflectionProperty::IS_PROTECTED : 0;
+        $modifiers += $node->isProtectedSet() ? ReflectionPropertyAdapter::IS_PROTECTED_SET_COMPATIBILITY : 0;
         $modifiers += $node->isPublic() ? CoreReflectionProperty::IS_PUBLIC : 0;
+        $modifiers += ($node->flags & ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY) === ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY ? ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY : 0;
+        $modifiers += ($node->flags & Modifiers::ABSTRACT) === Modifiers::ABSTRACT ? ReflectionPropertyAdapter::IS_ABSTRACT_COMPATIBILITY : 0;
 
+        /** @phpstan-ignore return.type */
         return $modifiers;
     }
+
+    private function computeImmediateVirtual(PropertyNode $node): bool
+    {
+        if ($node->hooks === []) {
```

Comment:
This block seems unnecessary - let's drop it, and we gain a removed array allocation too :)

Reply:
The block is necessary :) I have 19 failing tests without the code :D The logic after this code would have be more complicated without this if.